File: samples/csharp_dotnetcore/30.asp-mvc-bot/Models/ErrorViewModel.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

namespace Microsoft.BotBuilderSamples
{
    public class ErrorViewModel
    {
        public string RequestId { get; set; }

        public bool ShowRequestId => !string.IsNullOrEmpty(RequestId);
    }
}

File: archive/samples/javascript_nodejs/21.corebot-app-insights/models/errorViewModel.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

class ErrorViewModel {
    constructor(requestId = null) {
        this.requestId = requestId;
    }

    showRequestId() {
        return Boolean(this.requestId);
    }
}

module.exports.ErrorViewModel = ErrorViewModel;
